pkg/script/delegated_runner.go

```diff
@@ -0,0 +1,203 @@
+package script
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/kitt/progress"
+
+	"github.com/Azure/brigade/pkg/brigade"
+	"github.com/Azure/brigade/pkg/storage"
+	"github.com/Azure/brigade/pkg/storage/kube"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var logPattern = regexp.MustCompile(`\[brigade:k8s\]\s[a-zA-Z0-9-]+/[a-zA-Z0-9-]+ phase \w+`)
+
+const (
+	waitTimeout = 5 * time.Minute
+)
+
+func NewDelegatedRunner(c *kubernetes.Clientset, logDest io.Writer, namespace string, noProgress, background, verbose bool) (*scriptRunner, error) {
+	app := &scriptRunner{
+		store:          kube.New(c, namespace),
+		kc:             c,
+		namespace:      namespace,
+		logDestination: logDest,
+		noProgress:     noProgress,
+		background:     background,
+		verbose:        verbose,
+	}
+	return app, nil
+}
+
+type scriptRunner struct {
```

Comment:
Good point! I'll make change to export `ScriptRunner`.

Regarding `LogDestination`, would you mind if I renamed it to something like `PodLogDestination`, to avoid future collision? I was considering to redirect `fmt.Printf`s in the runner impl into another log destination, which will be named, perhaps, `RunnerLogDestination`. Btw, I appreciate suggestions for better names 😄 

Not sure about `Namespace`. I made the namespace private to keep nearly 1-on-1 mapping mapping between the original code and the refactored code, while avoiding a possible bug like namespaced pod getters use the namespace X and brigade store uses Y.

The namespace is used by two components, namespaced pod getters and brigade's store, each has different life cycles. The former is created on demand, whereas the latter is created on the creation of script runner. So exposing it may require us to make both components created on demand.